lib/liquid/optimizer.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Liquid
+  class Optimizer
+    def optimize(node)
+      case node
+      when Liquid::Template       then optimize(node.root.body)
+      when Liquid::Document       then optimize(node.body)
+      when Liquid::BlockBody      then optimize_block(node)
+      when Liquid::Variable       then optimize_variable(node)
+      when Liquid::Assign         then optimize_assign(node)
+      when Liquid::For            then optimize_for(node)
+      when Liquid::If             then optimize_if(node)
+      end
+      node
+    end
+
+    def optimize_block(block)
+      block.nodelist.each { |node| optimize(node) }
+    end
+
+    def optimize_variable(node)
+      # Turn chained `| append: "..."| append: "..."`, into a single `append_all: [...]`
+      if node.filters.size > 1 && node.filters.all? { |f, _| f == "append" }
+        node.filters = [["append_all", node.filters.map { |f, (arg)| arg }]]
+      end
```

Comment:
What about cases where the list of `append`s aren't the _entire_ list of filters, like `{{something | some_drop | append: "..." | append: "..." | another_drop }}`? Are they worth handling?

Reply:
Yeah! We could combine the consecutive one too. It's just more complex to handle.